src/pepr/operator/controllers/keycloak/authservice/authservice.ts

```diff
@@ -0,0 +1,120 @@
+import { Log, R } from "pepr";
+import { UDSConfig } from "../../../../config";
+import { Store } from "../../../common";
+import { UDSPackage } from "../../../crd";
+import { apiCall } from "../client-sync";
+import { Client } from "../types";
+import { updatePolicy } from "./authorization-policy";
+import { getAuthserviceConfig, operatorConfig, updateAuthServiceSecret } from "./config";
+import { Action, AuthServiceEvent, AuthserviceConfig, Chain } from "./types";
+
+export async function authservice(pkg: UDSPackage) {
+  // Get the list of clients from the package
+  const authServiceClients = R.filter(
+    sso => R.isNotNil(sso.enableAuthserviceSelector),
+    pkg.spec?.sso || [],
+  );
+
+  for (const client of authServiceClients) {
+    const name = `sso-client-${client.clientId}`;
+    const token = Store.getItem(name);
+    if (!token) {
+      throw new Error(`Failed to get token for client ${client.clientId}`);
+    }
+    const keycloakClient = await apiCall(client, "GET", token);
+
+    client.secret = keycloakClient.secret;
+    await reconcileAuthservice(
+      { name: client.clientId, action: Action.Add, client: client as unknown as Client },
+      client.enableAuthserviceSelector as { [key: string]: string },
```

Comment:
This is much clearer about what the assertion is:
```suggestion
      client.enableAuthserviceSelector!,
```
Also, if you just pass the `client` with the correct `Sso` type, you don't need to pass this argument separately anyway.

Reply:
Updated